### PR TITLE
Ben/csv

### PR DESCRIPTION
**Pull request by rouille**
### Purpose
Access the available profile on the server or locally using the `InputData` class.

### What the code is doing
Implement a `get_profile_version` in the `InputData` class that returns the profile version for the specified field (demand, hydro, solar or wind).

### Testing
```
>>> from powersimdata.input.input_data import InputData
>>> info = {"base_demand": "vJan2021", "grid_model": "usa_tamu"}
>>> id = InputData()
>>> version = id.get_profile_version(info, "demand")
>>> version
['vJan2021']
```
### Where to look
in the `powersimdata.input.input_data` module.

### Note
This function will be used in the `powersimdata.scenario.create` module to fetch the available profile version when the user builds a scenario. This is currently achieved by the `CSV` class. I did not touch the `powersimdata.scenario.create` module in this PR since we will refactor the `Create` and `_Builder` class in a next PR and any effort to remove the `CSV` instance and used the `get_profile_version` of the `InputData` class in place will most likely not be helpful for the new design.

### Time estimate
5 min

### CODE REVIEW

File: powersimdata/input/input_data.py [hunk 1]
```diff
@@ -25,6 +28,8 @@ def __init__(self, data_loc=None):
             "ct": "pkl",
             "grid": "mat",
         }
+        self.profile = {"demand", "hydro", "solar", "wind"}
```
Comment (jon-hagg) [inline powersimdata/input/input_data.py:31]: This feels like a module level constant

**Comment by jon-hagg** [comment]
Did you want the target to be the `model` branch? Looks like some changes here that were already reviewed and merged into that

**Comment by rouille** [comment]
> Did you want the target to be the `model` branch? Looks like some changes here that were already reviewed and merged into that

Good catch. Thanks

File: powersimdata/input/input_data.py (hunk 1 above)
Comment (rouille) [reply powersimdata/input/input_data.py:31]: Sounds good. I will declare it outside of the class

**Review by jon-hagg — APPROVED** [review]

File: powersimdata/input/input_data.py [hunk 2]
```diff
@@ -85,6 +87,33 @@ def get_data(self, scenario_info, field_name):
...
+            scenario_info["grid_model"],
+            kind + "_*",
+        )
+        stdin, stdout, stderr = self.data_access.execute_command("ls " + query)
+        if len(stderr.readlines()) != 0:
+            print("No %s profiles available." % kind)
+            version = []
+        else:
+            filename = [os.path.basename(line.rstrip()) for line in stdout.readlines()]
+            version = [f[f.rfind("_") + 1 : -4] for f in filename]
```
Comment (danielolsen) [inline powersimdata/input/input_data.py:114]: Do we ever have the case where there is more than one filename on the same line? In this case, does the parsing of the `ls` output still work properly?

Comment (rouille) [reply powersimdata/input/input_data.py:114]: never thought about it. It seems that on Unix like system `ls` returns values on separate lines, see example below:
```
(PowerSimData) [~/CEM/PowerSimData] (ben/csv) brdo$ ls
ATTRIBUTION.md            Pipfile                   docker-compose.yml        powersimdata.egg-info     setup.py
Dockerfile                Pipfile.lock              optional-requirements.txt pytest.ini                tox.ini
LICENSE                   README.md                 powersimdata              requirements.txt
(PowerSimData) [~/CEM/PowerSimData] (ben/csv) brdo$ ls | wc -l
      14
```

File: powersimdata/input/input_data.py (hunk 1 above)
Comment (rouille) [reply powersimdata/input/input_data.py]: Done

File: powersimdata/input/input_data.py (hunk 2 above)
Comment (danielolsen) [reply powersimdata/input/input_data.py:111]: Cool, I see the same thing when I'm ssh'ed in, I just wasn't sure whether your example would be parsed as three lines or as fourteen.

**Review by danielolsen — APPROVED** [review]
Looks good.

**Pull request merged**